flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/optimize/MiniBatchOptimizationTest.java

```diff
@@ -0,0 +1,169 @@
+                        + "  c VARCHAR,\n"
+                        + "  d BIGINT\n"
+                        + ") WITH (\n"
+                        + "  'connector' = 'values',\n"
+                        + "  'bounded' = 'false')");
+    }
+
+    private boolean containsMiniBatch(String sql) {
+        final Table result = streamTableEnv.sqlQuery(sql);
+        RelNode relNode = TableTestUtil.toRelNode(result);
+        StreamPlanner planner =
+                (StreamPlanner) ((TableEnvironmentImpl) streamTableEnv).getPlanner();
+        StreamCommonSubGraphBasedOptimizer optimizer =
+                new StreamCommonSubGraphBasedOptimizer(planner);
+        Seq<RelNode> nodeSeq =
+                JavaConverters.asScalaIteratorConverter(Arrays.asList(relNode).iterator())
+                        .asScala()
+                        .toSeq();
+        Seq<RelNodeBlock> blockSeq = optimizer.doOptimize(nodeSeq);
+        List<RelNodeBlock> blockList = scala.collection.JavaConverters.seqAsJavaList(blockSeq);
+        boolean res =
+                blockList.stream()
+                        .map(
+                                b ->
+                                        !b.getMiniBatchInterval()
+                                                .equals(
+                                                        MiniBatchIntervalTrait.NONE()
+                                                                .getMiniBatchInterval()))
+                        .reduce(false, (l, r) -> l || r);
+        return res;
+    }
+
+    @TestTemplate
+    public void testMiniBatchWithAggregation() {
+        final String aggQuery =
+                "SELECT\n"
+                        + "  AVG(a) AS avg_a,\n"
+                        + "  COUNT(*) AS cnt,\n"
+                        + "  count(b) AS cnt_b,\n"
+                        + "  min(b) AS min_b,\n"
+                        + "  MAX(c) FILTER (WHERE a > 1) AS max_c\n"
+                        + "FROM MyTableA";
+
+        boolean containsMiniBatch = containsMiniBatch(aggQuery);
+        if (isMiniBatchEnabled) {
+            assertTrue(containsMiniBatch);
+        } else {
+            assertFalse(containsMiniBatch);
+        }
+    }
+
+    @TestTemplate
+    public void testMiniBatchWithJoin() {
+        final String joinQuery = "SELECT * FROM MyTableA a, MyTableB b WHERE a.a = b.a";
+
+        boolean containsMiniBatch = containsMiniBatch(joinQuery);
+        if (isMiniBatchEnabled) {
+            assertTrue(containsMiniBatch);
+        } else {
+            assertFalse(containsMiniBatch);
+        }
+    }
+
+    @TestTemplate
+    public void testMiniBatchWithProjectFilter() {
+        final String joinQuery = "SELECT b FROM MyTableA a WHERE a.a > 123";
+
+        boolean containsMiniBatch = containsMiniBatch(joinQuery);
+        assertFalse(containsMiniBatch);
+    }
+
+    @Parameters(name = "isMiniBatchEnabled={0}, miniBatchLatency={1}, miniBatchSize={2}")
+    public static Object[][] data() {
```

Comment:
The minibatch latency and minibatch size are no need to be part of `Parameters`, right?